homeassistant/components/escea/climate.py

```diff
@@ -0,0 +1,217 @@
+        HVAC_MODE_OFF,
+    ]
+    _attr_icon = ICON
+    _attr_precision = PRECISION_WHOLE
+    _attr_should_poll = False
+    _attr_supported_features = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+    _attr_target_temperature_step = PRECISION_WHOLE
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, controller: Controller) -> None:
+        """Initialise ControllerDevice."""
+        self._controller = controller
+
+        self._attr_min_temp = controller.min_temp
+        self._attr_max_temp = controller.max_temp
+
+        self._attr_unique_id = controller.device_uid
+        self._attr_name = f"Escea Fireplace {self._attr_unique_id}"
+
+        self._attr_available = True
+
+    async def async_added_to_hass(self):
+        """Call on adding to hass.
+
+        Registers for connect/disconnect/update events
+        """
+
+        @callback
+        def controller_disconnected(ctrl: Controller, ex: Exception) -> None:
+            """Disconnected from controller."""
+            if ctrl is not self._controller:
+                return
+            self.set_available(False, ex)
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, DISPATCH_CONTROLLER_DISCONNECTED, controller_disconnected
+            )
+        )
+
+        @callback
+        def controller_reconnected(ctrl: Controller) -> None:
+            """Reconnected to controller."""
+            if ctrl is not self._controller:
+                return
+            self.set_available(True)
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, DISPATCH_CONTROLLER_RECONNECTED, controller_reconnected
+            )
+        )
+
+        @callback
+        def controller_update(ctrl: Controller) -> None:
+            """Handle controller data updates."""
+            if ctrl is not self._controller:
+                return
+            self.async_write_ha_state()
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, DISPATCH_CONTROLLER_UPDATE, controller_update
+            )
+        )
+
+    @callback
+    def set_available(self, available: bool, ex: Exception = None) -> None:
+        """Set availability for the controller."""
+        if self._attr_available == available:
+            return
+
+        if available:
+            _LOGGER.info("Reconnected controller %s ", self._controller.device_uid)
+        else:
+            _LOGGER.info(
+                "Controller %s disconnected due to exception: %s",
+                self._controller.device_uid,
+                ex,
+            )
+
+        self._attr_available = available
+        self.async_write_ha_state()
+
+    @property
+    def device_info(self):
+        """Return the device info for the Escea system."""
+        return {
+            "identifiers": {(ESCEA, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Escea",
+        }
```

Comment:
This can be set as `self._attr_device_info`

Also instead of a dict use `DeviceInfo`

Reply:
Fixed - thanks